atum/src/main/scala/za/co/absa/atum/core/MeasurementProcessor.scala

```diff
@@ -17,23 +17,109 @@ package za.co.absa.atum.core
 
 import org.apache.spark.sql.functions._
 import org.apache.spark.sql.types.{DecimalType, LongType, StringType}
-import org.apache.spark.sql.{Column, Dataset, Row}
+import org.apache.spark.sql.{Column, DataFrame, Dataset, Row}
 import za.co.absa.atum.core.ControlType._
+import za.co.absa.atum.core.MeasurementProcessor.MeasurementFunction
 import za.co.absa.atum.model.Measurement
 import za.co.absa.atum.utils.controlmeasure.ControlMeasureUtils
 
+import scala.util.{Failure, Success, Try}
+
+object MeasurementProcessor {
+  type MeasurementFunction = DataFrame => String
+
+  private val valueColumnName: String = "value"
+
+  /**
+   * Assembles the measurement function based for controlCol based on the controlType
+   * @param controlCol
+   * @param controlType
+   * @return
+   */
+  private[atum] def getMeasurementFunction(controlCol: String, controlType: ControlType): MeasurementFunction = {
+    controlType match {
+      case Count => (ds: Dataset[Row]) => ds.count().toString
+      case DistinctCount => (ds: Dataset[Row]) => {
+        ds.select(col(controlCol)).distinct().count().toString
+      }
+      case AggregatedTotal =>
+        (ds: Dataset[Row]) => {
+          val aggCol = sum(col(valueColumnName))
+          aggregateColumn(ds, controlCol, aggCol)
+        }
+      case AbsAggregatedTotal =>
+        (ds: Dataset[Row]) => {
+          val aggCol = sum(abs(col(valueColumnName)))
+          aggregateColumn(ds, controlCol, aggCol)
+        }
```

Comment:
We do -- in `ControlMeasureBuilder.deriveControlType()`. There, if `Specific` strategy is used and `controlType == AggregatedTotal || controlType == AbsAggregatedTotal`, the column type is checked to be numeric. We issue a warning in such a case.

Reply:
Not sure if it is not too high in the hierarchy
Also, is only warning enough? Do we know what happens if it is misused?

Reply:
By _not too high in the hierarchy_, you mean that the warning should not occur when `calculateMeasurement()` is called and the controlType gets derived then, but right when the user attempts to set it? It can be done, it is just a bit unfortunate that since there are multiple API options for this, there may need to be multiple places to perform the check (at least in `private def withAggregateColumn` and `private def withAggregateColumnsDirectly`) -- but sure, it would make sense I guess.

As for the warning vs error, in that case, I am slightly siding with a warning, but I let myself be persuaded. I can imagine weird scenarios where perhaps some obscure experimental usage might make sense on non-numeric fields, too (Binary, TimeStamp, ...), so I don't want to restrict it right away.

Reply:
I have moved the check "higher", now it checks the datatype vs controlType numericity on the builder methods (`.withAggregateColumn(s)`). I left it as a warning for now.